scheduler/extender/util.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// logNodes prints a line for every node.
+func logNodes(nodes *api.NodeList) {
+	for _, n := range nodes.Items {
+		fmt.Printf("Received node %v with joules %v\n", n.Name, n.Labels["joules"])
+	}
+}
+
+// selectNode returns the one node with the lowest joules label value out
+// of a list of nodes.
+func selectNode(nodes *api.NodeList) (*api.Node, error) {
```

Comment:
Why don't you return a []Node{} as this is what you will eventually have to send back to k8s scheduler?